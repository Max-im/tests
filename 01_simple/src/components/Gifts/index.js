import React, { Component } from "react";
import {
  Form,
  FormGroup,
  FormControl,
  ControlLabel,
  Button
} from "react-bootstrap";

export default class Gifts extends Component {
  state = {
    person: "",
    present: ""
  };

  render() {
    return (
      <div>
        <Form>
          <FormGroup>
            <label>
              Person
              <FormControl
                onChange={e => this.setState({ person: e.target.value })}
                className="input-person"
              />
            </label>
          </FormGroup>
          <FormGroup>
            <label>
              Present
              <FormControl
                onChange={e => this.setState({ present: e.target.value })}
                className="input-present"
              />
            </label>
          </FormGroup>

          <Button
            className="btn-remove"
            onClick={() => {
              this.props.removeGift(this.props.gift.id);
            }}
          >
            Remove gift
          </Button>
        </Form>
      </div>
    );
  }
}
