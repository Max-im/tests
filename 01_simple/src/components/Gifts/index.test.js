import React from "react";
import Enzyme, { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Gifts from "./";
Enzyme.configure({ adapter: new Adapter() });

describe("Gift", () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = { gift: { id }, removeGift: mockRemove };
  const gift = shallow(<Gifts {...props} />);

  /*
   * Render correctly
   */
  test("render correctly", () => {
    const component = renderer.create(<Gifts />).toJSON();
    expect(component).toMatchSnapshot();
  });

  /*
   * Init person and present in state
   */
  test("Init person and present in state", () => {
    expect(gift.state()).toEqual({ person: "", present: "" });
  });

  /*
   * Typing into person input
   */
  describe("Fill out person input", () => {
    const person = "Uncle";
    beforeEach(() => {
      gift
        .find(".input-person")
        .simulate("change", { target: { value: person } });
    });

    test("updates the person in state", () => {
      expect(gift.state().person).toBe(person);
    });
  });

  /*
   * Typing into present input
   */
  describe("Fill out person input", () => {
    const present = "Golf club";
    beforeEach(() => {
      gift
        .find(".input-present")
        .simulate("change", { target: { value: present } });
    });

    test("updates the present in state", () => {
      expect(gift.state().present).toBe(present);
    });
  });

  /*
   * Clicking a remove gift btn
   */
  describe("Clicking a remove gift btn", () => {
    beforeEach(() => gift.find(".btn-remove").simulate("click"));

    // remove btn click callback
    test("remove btn click callback", () => {
      expect(mockRemove).toHaveBeenCalledWith(id);
    });
  });
});
