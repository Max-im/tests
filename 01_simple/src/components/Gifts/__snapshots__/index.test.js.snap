// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift render correctly 1`] = `
<div>
  <form
    className=""
  >
    <div
      className="form-group"
    >
      <label>
        Person
        <input
          className="input-person form-control"
          onChange={[Function]}
        />
      </label>
    </div>
    <div
      className="form-group"
    >
      <label>
        Present
        <input
          className="input-present form-control"
          onChange={[Function]}
        />
      </label>
    </div>
    <button
      className="btn-remove btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Remove gift
    </button>
  </form>
</div>
`;
